kornia/augmentation/_2d/base.py

```diff
@@ -70,7 +70,17 @@ def compute_transformation(self, input: Tensor, params: Dict[str, Tensor], flags
 
     def generate_transformation_matrix(self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]) -> Tensor:
         """Generate transformation matrices with the given input and param settings."""
-        to_apply = params['batch_prob']
+        # Note about the explicit type conversions here:
+        # If this code is run in an autocast-enabled region, the transformation matrix or the output
+        # may be float16 even though the input was float32 (e.g. torch.mm produces float16 output),
+        # see also the documentation on autocasting: https://pytorch.org/docs/stable/amp.html.
+        # It may be unexpected for the user if the output type changes and it can also lead to errors
+        # for the index_put operation below (see also https://github.com/kornia/kornia/issues/1737)
+        # In case the type already matches the input type, the conversions are no-ops
```

Comment:
i removed this because is done in different places in the same way -- also now with the `if is_autocast_enabled` it's self-explanatory
```suggestion
```